File: src/main/java/kitchenpos/menus/tobe/domain/MenuGroup.java
package kitchenpos.menus.tobe.domain;

import jakarta.persistence.*;

import java.util.UUID;

@Table(name = "menu_group")
@Entity(name = "tobeMenuGroup")
public class MenuGroup {
    @Column(name = "id", columnDefinition = "binary(16)")
    @Id
    private UUID id;

    @Embedded
    private MenuGroupName name;

    protected MenuGroup() {
    }

    public MenuGroup(final UUID id, final MenuGroupName name) {
        this.id = id;
        this.name = name;
    }

    public UUID getId() {
        return id;
    }
}

File: src/main/java/kitchenpos/menus/tobe/domain/MenuGroup.kt
package kitchenpos.menus.tobe.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.*

@Table(name = "menu_group")
@Entity
class MenuGroup(
    @Column(name = "name", nullable = false)
    val name: MenuGroupName
) {
    @Column(name = "id", columnDefinition = "binary(16)")
    @Id
    val id: UUID = UUID.randomUUID()
}
